packages/wrangler/src/__tests__/r2.test.ts

```diff
@@ -196,12 +199,123 @@ describe("r2", () => {
 				);
 				await runWrangler("r2 bucket create testBucket -J eu");
 				expect(std.out).toMatchInlineSnapshot(`
-				            "Creating bucket testBucket (eu).
-				            Created bucket testBucket (eu)."
+				            "Creating bucket testBucket (eu) with default storage class set to Standard.
+				            Created bucket testBucket (eu) with default storage class set to Standard."
+			          `);
+			});
+
+			it("should create a bucket with the expected default storage class", async () => {
+				msw.use(
+					rest.post(
+						"*/accounts/:accountId/r2/buckets",
+						async (request, response, context) => {
+							const { accountId } = request.params;
+							expect(accountId).toEqual("some-account-id");
+							expect(request.headers.get("cf-r2-storage-class")).toEqual(
```

Comment:
I know we do this all over the place, but really we should not be using `expect` checks inside the mock service worker handlers.
Instead, we should simulate what the backend would do, which I assume would be some kind of 4xx response in the case that the storage class is invalid?

Then in these tests we can simply check that Wrangler does not error itself, which proves that it must have passed the correct header.
Equally we can check that Wrangler handles the error response correctly if the header is invalid.

Reply:
Good point, going to update the test.